### DISCUSSION TITLE
Feature Request: Please include code examples with full components like MUI docs

### DISCUSSION THREAD

**Discussion by kpervin** [mantinedev/mantine#456] [Feedback]
When you go to the Material UI docs, one thing their docs do super well is that they have an expanded view to see the full source code for a [component's example](https://mui.com/components/autocomplete/#combo-box). This includes imports and exports, as well as all composite components created in the example. It would be great to get something like this in Mantine as well.

**Comment by rtivital** [comment]
Hi @kpervin, we are trying to build documentation the most accessible way possible. It means that we usually omit unnecessary details from demos code to showcase specific feature. Where it is necessary, we, of course, provide full code. I think that this is the best way to document things, and it saves both ours and yours time.

**Reply by rtivital** [reply]
And if you can provide several examples that are not clear for you, we will consider to update/replace them, so please share what confuses you in docs.

**Reply by kpervin** [reply]
Ok! So far, I'm loving what I'm seeing of this project. Been using Material UI for a long time, but this looks very promising mainly because you include both the `sx` prop AND `useStyles`. Their removal of `useStyles` was a bit of a sore spot for me, so this looks great! Please keep up the good work :)

**Reply by kpervin** [reply]
I think the only thing that I miss in comparison to the two doc styles are seeing imports.

**Reply by rtivital** [reply]
Mantine does not have separate packages for each component, you can import almost everything from @mantine/core or @mantine/dates
